app/bundles/ReportBundle/Tests/Model/ReportModelTest.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace Mautic\ReportBundle\Tests\Model;
+
+use Mautic\ChannelBundle\Helper\ChannelListHelper;
+use Mautic\CoreBundle\Helper\CoreParametersHelper;
+use Mautic\CoreBundle\Helper\TemplatingHelper;
+use Mautic\LeadBundle\Model\FieldModel;
+use Mautic\ReportBundle\Helper\ReportHelper;
+use Mautic\ReportBundle\Model\CsvExporter;
+use Mautic\ReportBundle\Model\ExcelExporter;
+use Mautic\ReportBundle\Model\ReportModel;
+use Mautic\ReportBundle\Tests\Fixtures;
+
+class ReportModelTest extends \PHPUnit\Framework\TestCase
+{
+    /**
+     * @var ReportModel
+     */
+    private $reportModel;
+
+    protected function setUp()
+    {
+        $this->reportModel = new ReportModel(
+            $this->getMockBuilder(CoreParametersHelper::class)->disableOriginalConstructor()->getMock(),
+            $this->getMockBuilder(TemplatingHelper::class)->disableOriginalConstructor()->getMock(),
+            $this->getMockBuilder(ChannelListHelper::class)->disableOriginalConstructor()->getMock(),
+            $this->getMockBuilder(FieldModel::class)->disableOriginalConstructor()->getMock(),
+            $this->getMockBuilder(ReportHelper::class)->disableOriginalConstructor()->getMock(),
+            $this->getMockBuilder(CsvExporter::class)->disableOriginalConstructor()->getMock(),
+            $this->getMockBuilder(ExcelExporter::class)->disableOriginalConstructor()->getMock()
+        );
+    }
+
+    public function testGetColumnListWithContext()
+    {
+        $this->reportModel->reportBuilderData = Fixtures::getReportBuilderEventData();
+
+        $actual   = $this->reportModel->getColumnList('assets');
```

Comment:
This test does not explain why the added code `explode('.', $context);` is needed. Could you please also add a test where that dot is in the `$context`?

Reply:
The bug has do with the format of the array of available reports having changed. The instance where it occurred is when the reportModel->getColumn method was called. This test covers that method, which calls buildAvailableReports, where the fix occurs. 

The problem wasn't that a dot got added to the context, but that the whole array changed.